gosnmp.go

```diff
@@ -181,6 +186,12 @@ func (x *GoSNMP) Connect() error {
 		x.loggingEnabled = true
 	}
 
+	if x.MaxOids == 0 {
+		x.MaxOids = maxOids
+	} else if x.MaxOids < 0 {
+		return fmt.Errorf("MaxOids cannot be less than 0")
+	}
+
```

Comment:
This code checks the variable MaxOids is zero:
if it is, assume it's a default zero value and set our default
if it's less than zero, return an error, since someone has manually set this and should know that it's invalid
else continue, as the user has set a valid (non-zero) value